File: packages/terminal/ios/Plugin/StripeTerminalPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(StripeTerminalPlugin, "StripeTerminal",
           CAP_PLUGIN_METHOD(initialize, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(setConnectionToken, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(discoverReaders, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(cancelDiscoverReaders, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(connectReader, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(getConnectedReader, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(disconnectReader, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(collectPaymentMethod, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(cancelCollectPaymentMethod, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(confirmPaymentIntent, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(setSimulatorConfiguration, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(installAvailableUpdate, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(cancelInstallUpdate, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(setReaderDisplay, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(clearReaderDisplay, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(rebootReader, CAPPluginReturnPromise);
)

File: packages/terminal/ios/Plugin/StripeTerminalPlugin.swift
import Foundation
import StripeTerminal
import Capacitor
import PassKit

@objc(StripeTerminalPlugin)
public class StripeTerminalPlugin: CAPPlugin {
    private let implementation = StripeTerminal()

    override public func load() {
        super.load()
        self.implementation.plugin = self
        // TODO: add STPAPIClient.shared.appInfo
    }

    @objc func initialize(_ call: CAPPluginCall) {
        self.implementation.initialize(call)
    }

    @objc func setConnectionToken(_ call: CAPPluginCall) {
        self.implementation.setConnectionToken(call)
    }

    @objc func discoverReaders(_ call: CAPPluginCall) {
        do {
            try self.implementation.discoverReaders(call)
        } catch {
            call.reject("discoverReaders throw error.")
        }
    }

    @objc func setSimulatorConfiguration(_ call: CAPPluginCall) {
        self.implementation.setSimulatorConfiguration(call)
    }

    @objc func cancelDiscoverReaders(_ call: CAPPluginCall) {
        self.implementation.cancelDiscoverReaders(call)
    }

    @objc func connectReader(_ call: CAPPluginCall) {
        self.implementation.connectReader(call)
    }

    @objc func getConnectedReader(_ call: CAPPluginCall) {
        self.implementation.getConnectedReader(call)
    }

    @objc func disconnectReader(_ call: CAPPluginCall) {
        self.implementation.disconnectReader(call)
    }

    @objc func collectPaymentMethod(_ call: CAPPluginCall) {
        self.implementation.collectPaymentMethod(call)
    }

    @objc func cancelCollectPaymentMethod(_ call: CAPPluginCall) {
        self.implementation.cancelCollectPaymentMethod(call)
    }

    @objc func confirmPaymentIntent(_ call: CAPPluginCall) {
        self.implementation.confirmPaymentIntent(call)
    }

    @objc func installAvailableUpdate(_ call: CAPPluginCall) {
        self.implementation.installAvailableUpdate(call)
    }

    @objc func cancelInstallUpdate(_ call: CAPPluginCall) {
        self.implementation.cancelInstallUpdate(call)
    }

    @objc func setReaderDisplay(_ call: CAPPluginCall) {
        self.implementation.setReaderDisplay(call)
    }

    @objc func clearReaderDisplay(_ call: CAPPluginCall) {
        self.implementation.clearReaderDisplay(call)
    }

    @objc func rebootReader(_ call: CAPPluginCall) {
        self.implementation.rebootReader(call)
    }

    @objc func cancelReaderReconnection(_ call: CAPPluginCall) {
        self.implementation.cancelReaderReconnection(call)
    }

}
